src/coreclr/jit/importer.cpp

```diff
@@ -9602,7 +9602,16 @@ var_types Compiler::impImportCall(OPCODE                  opcode,
         {
             return impImportJitTestLabelMark(sig->numArgs);
         }
-#endif // DEBUG
+
+        // static ulong JitHelpers_JitFlags() => 0;
+        // can be defined anywhere and will be replaced by Debug-version of RyuJIT
+        if ((mflags & CORINFO_FLG_STATIC) && (sig->numArgs == 0) && (sig->retType == CorInfoType::CORINFO_TYPE_ULONG) &&
+            (strcmp("JitHelpers_JitFlags", eeGetMethodName(methHnd, nullptr)) == 0))
+        {
+            call = gtNewLconNode((__int64)opts.jitFlags->GetRawFlags());
+            goto DONE_CALL;
+        }
+#endif
```

Comment:
Should we only recognize this "intrinsic" under some config setting? Or else make it into a real / official intrinsic?

Reply:
I'll prepare a public api proposal

Reply:
Can we use tracing for this instead of intrinsics like this?

We need to be able to replay the sequence of tiered compilations, without instrumenting user code.

Reply:
@jkotas I needed this intrinsic to make sure the correct version of code was applied and **is executed**, it helped me to catch a bug: everything looked fine on the traces level, JitDisasm also displayed codegen for all code versions but it turns out one of them was not applied because I forgot to update EntryPoint (it was in the initial impl where I was patching existing call counting stubs instead of creating new ones) - I can delete this one now

Reply:
You should be able to figure out using tracing what kind of code is executing.

Reply:
Makes sense, I don't really need this now so will remove once the outerloop tests finish